homeassistant/components/notify/yessssms.py

```diff
@@ -38,14 +39,35 @@ def __init__(self, username, password, recipient):
         from YesssSMS import YesssSMS
         self.yesss = YesssSMS(username, password)
         self._recipient = recipient
+        # pylint: disable=protected-access
+        _LOGGER.debug(
+            "initialized; library version: %s", self.yesss._version)
 
     def send_message(self, message="", **kwargs):
         """Send a SMS message via Yesss.at's website."""
+        if self.yesss.account_is_suspended():
+            # only retry to login after HASS was restarted with (hopefully)
+            # new login data.
+            _LOGGER.critical(
+                "Account is suspended, cannot send SMS. "
+                "Check your login data!")
+            return
         try:
             self.yesss.send(self._recipient, message)
-        except ValueError as ex:
-            if str(ex).startswith("YesssSMS:"):
-                _LOGGER.error(str(ex))
-        except RuntimeError as ex:
-            if str(ex).startswith("YesssSMS:"):
-                _LOGGER.error(str(ex))
+        except self.yesss.NoRecipientError as ex:
+            _LOGGER.error(
+                "You need to provide a recipient for SMS notification: %s", ex)
```

Comment:
Instead of using `%s` to pass in the exception text, pass the full exception object to the logger as shown below. This allows the logger to print the message as well as the full stack trace, which is very helpful for debugging.
```python
_LOGGER.error(
    "You need to provide a recipient for SMS notification", exc_info=ex)
```